src/Channel.php

```diff
@@ -97,6 +105,16 @@ public function events(): Events
         return $this->events;
     }
 
+    private function assertOpen(): void {
+        if ($this->state !== self::STATE_OPEN) {
+            throw Exception\ChannelException::notOpen($this->id);
+        }
+    }
+
+    public function isOpen(): bool {
```

Comment:
Just convenient when we want to check if a channel is still alive. I used it in consumers because between an arriving message + ack/nack, the channel can be closed - which is very likely e.g. if a server dies.
To not run into unnecessary exceptions, i check for "isOpen" and only handle other unexpected exceptions when acking